pkg/client/preflight.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2018 Heptio Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"encoding/json"
+	"fmt"
+
+	version "github.com/hashicorp/go-version"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apimachineryversion "k8s.io/apimachinery/pkg/version"
+	"k8s.io/client-go/kubernetes"
+)
+
+var preflightChecks = []func(kubernetes.Interface) error{
+	preflightDNSCheck,
+	preflightVersionCheck,
+}
+
+// PreflightChecks runs all preflight checks in order, returning the first error encountered.
```

Comment:
what was the decision behind bailing on the first one and not collecting a list of errors?

Reply:
I guess it's not obvious to me how to return a list of errors. There's no ErrorList that implements Error() is there?

Reply:
I was thinking []errors a la kubeadm https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/preflight/checks.go

Reply:
We'd need to thread that all the way back to client.Interface, is the issue